server/action_creators.js

```diff
@@ -70,6 +72,36 @@ export function addOrder(entries) {
   };
 }
 
+export function changeEntryStatus(orderId, entryId, status) {
```

Comment:
This is miserable.. I'm sure some of the fetching patterns could be dried up a bit.. or maybe I'm being dumb about it, but I think we should really look into using Immutable or some other functional utility library to handle these deep accesses if we want to maintain our purely nondestructive updates.